flake.nix

```diff
@@ -23,15 +23,10 @@
     , ...
     } @ inputs:
     let
-      nameValuePair = name: value: { inherit name value; };
-      genAttrs = names: f: builtins.listToAttrs (map (n: nameValuePair n (f n)) names);
-      allSystems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
+      supportedSystems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
 
-      forAllSystems = f: genAttrs allSystems (system: f rec {
```

Comment:
I don't necessarily agree with removing `pkgs` and friends from this (I like having access to `system` and `pkgs` from the arguments rather than having to bind `pkgs` using `nixpkgsFor`), but it doesn't functionally change anything, so :shrug: